Add documentation for secret, configmap and downwardAPI file permissions for 1.4 release

rata opened the pull request:
The patch that adds this feature in core kubernetes,
https://github.com/kubernetes/kubernetes/pull/28936, was merged and will be
released with kubernetes 1.4.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1178)
<!-- Reviewable:end -->

rata commented:
The test fail seems unrelated. It says: `examples_test.go:293: ../docs/user-guide/sj.yaml: sj does not have a test case defined`

rata commented:
And this seems to be the fixed, so probably this will pass once this other PR is merged: https://github.com/kubernetes/kubernetes.github.io/pull/1159

devin-donnelly commented:
Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions.

---

*[docs/user-guide/configmap/index.md, line 417 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178#-KRAGi5alZ7dMTTj2K7_:-KRAGi5alZ7dMTTj2K7a:bin67gx) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/b547bc46f45447937700b281e2118462db86f599/docs/user-guide/configmap/index.md#L417)):*
> ```Markdown
> ```
> 
> #### Projections of keys to specific paths and file permissions
> ```

In general, prefer active to passive voice.

"Projecting keys to specific paths and file permissions"

---

*[docs/user-guide/configmap/index.md, line 419 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178#-KRAGqJIOesmrSKaWDhB:-KRAGqJIOesmrSKaWDhC:bqe7pma) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/b547bc46f45447937700b281e2118462db86f599/docs/user-guide/configmap/index.md#L419)):*
> ```Markdown
> #### Projections of keys to specific paths and file permissions
> 
> Projections of keys to specific paths and specific permissions per file are also
> ```

Same w/r/t active/passive voice.

"You can project keys to specific paths and specific permissions on a per-file basis. The [Secrets](/docs/user-guide/secrets/) user guide explains the syntax."

---

*[docs/user-guide/downward-api/index.md, line 117 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178#-KRAH9mLWqk2Pr71hji2:-KRAH9mLWqk2Pr71hji3:b-nzje2c) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/b547bc46f45447937700b281e2118462db86f599/docs/user-guide/downward-api/index.md#L117)):*
> ```Markdown
> In future, it will be possible to specify a specific annotation or label.
> 
> #### Projections of keys to specific paths and file permissions
> ```

"Projecting keys to specific paths and file permissions"

---

*[docs/user-guide/downward-api/index.md, line 119 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178#-KRAH4r20O7CzZG-_v8d:-KRAH4r20O7CzZG-_v8e:b-fljzl7) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/b547bc46f45447937700b281e2118462db86f599/docs/user-guide/downward-api/index.md#L119)):*
> ```Markdown
> #### Projections of keys to specific paths and file permissions
> 
> Projections of keys to specific paths and specific permissions per file are also
> ```

"You can project keys to specific paths and specific permissions on a per-file basis. The [Secrets](/docs/user-guide/secrets/) user guide explains the syntax."

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178)*
<!-- Sent from Reviewable.io -->

devin-donnelly commented:
Some minor wording changes; once cleaned up, I'll add Docs LGTM.

rata commented:
Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions.

---

*[docs/user-guide/configmap/index.md, line 417 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178#-KRAGi5alZ7dMTTj2K7_:-KRAqbS6dJWqak9XC64L:bz8jsh3) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/b547bc46f45447937700b281e2118462db86f599/docs/user-guide/configmap/index.md#L417)):*
<details><summary><i>Previously, devin-donnelly wrote…</i></summary>
> In general, prefer active to passive voice.
> 
> "Projecting keys to specific paths and file permissions"
</details>
Fixed! :)

The title was based on this other from the "secrets" section: "Projection of secret keys to specific paths"

---

*[docs/user-guide/configmap/index.md, line 419 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178#-KRAGqJIOesmrSKaWDhB:-KRArCLQS6V_5rKdJxqt:bt2vr0b) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/b547bc46f45447937700b281e2118462db86f599/docs/user-guide/configmap/index.md#L419)):*
<details><summary><i>Previously, devin-donnelly wrote…</i></summary>
> Same w/r/t active/passive voice.
> 
> "You can project keys to specific paths and specific permissions on a per-file basis. The [Secrets](/docs/user-guide/secrets/) user guide explains the syntax."
</details>
Fixed

---

*[docs/user-guide/downward-api/index.md, line 117 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178#-KRAH9mLWqk2Pr71hji2:-KRAreleTj559JS2aKuf:b-lx4fuw) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/b547bc46f45447937700b281e2118462db86f599/docs/user-guide/downward-api/index.md#L117)):*
<details><summary><i>Previously, devin-donnelly wrote…</i></summary>
> "Projecting keys to specific paths and file permissions"
</details>
Fixed!

---

*[docs/user-guide/downward-api/index.md, line 119 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178#-KRAH4r20O7CzZG-_v8d:-KRArhAHa1bY8FRjbs33:bt2vr0y) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/b547bc46f45447937700b281e2118462db86f599/docs/user-guide/downward-api/index.md#L119)):*
<details><summary><i>Previously, devin-donnelly wrote…</i></summary>
> "You can project keys to specific paths and specific permissions on a per-file basis. The [Secrets](/docs/user-guide/secrets/) user guide explains the syntax."
</details>
Fixed!

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178)*
<!-- Sent from Reviewable.io -->

rata commented:
Thanks, I think I addressed them all. Please take another look :)

---

Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/1178#-:-KRAsA_zymr7gHO4CiT0:byfyc1i)*
<!-- Sent from Reviewable.io -->

devin-donnelly commented:
@rata , would you mind rebasing this PR on the latest from release-1.4? I'm testing some staging things.

rata commented:
@devin-donnelly sure, np! Rebased now :)

devin-donnelly commented:
Thanks! And Docs LGTM. Still going to hold for Tech Review; I'll flag it.

rata commented:
@devin-donnelly thanks! I wrote the core kubernetes patch (https://github.com/kubernetes/kubernetes/pull/28936) adding this functionality, so maybe there is no need to bother someone else for the tech review.

But I understand if someone from the kubernetes org needs to confirm. Just saying in case we can avoid @bgrant0607's time! :)

devin-donnelly commented:
Oh, okay. No problem, then.
